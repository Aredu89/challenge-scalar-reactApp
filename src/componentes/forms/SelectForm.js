import React from 'react';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormHelperText from '@material-ui/core/FormHelperText';
import Spinner from '../common/Spinner'
import { makeStyles } from '@material-ui/core/styles';
import './SelectForm.css';

const useStyles = makeStyles((theme) => ({
  formControl: {
    margin: theme.spacing(1),
    width: "100%",
  },
  selectEmpty: {
    marginTop: theme.spacing(2),
  },
}));

const SelectForm = props =>{
  //options: Array con formato [{ value: 10, label: "Diez" }]
  const {
    label,
    value,
    options,
    onChange,
    error,
    errorText,
    loading
  } = props
  const classes = useStyles();

  return(
    <div className="select-form w-100">
      {
        loading ?
        <Spinner />
        :
        <FormControl
          variant="outlined"
          className={classes.formControl}
          error={error}
          >
          <InputLabel id={`select-${label}`}>{label ? label : ""}</InputLabel>
          <Select
            labelId={`select-${label}`}
            id={`select-${value}`}
            value={value ? value : ""}
            onChange={onChange}
            label={label ? label : ""}
          >
            <MenuItem value={null}>Sin seleccionar</MenuItem>
            {
              options ?
              options.length > 0 ?
              options.map((option, i)=>{
                return(
                  <MenuItem key={i} value={option.value}>{option.label}</MenuItem>
                )
              })
              : null
              : null
            }
          </Select>
          {
            error && errorText &&
            <FormHelperText>{errorText}</FormHelperText>
          }
        </FormControl>
      }
    </div>
  )
}

export default SelectForm