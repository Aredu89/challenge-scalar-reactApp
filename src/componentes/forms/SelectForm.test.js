import React from 'react';
import { render, cleanup } from '@testing-library/react';
import SelectForm from './SelectForm';

afterEach(cleanup)

describe('SelectForm tests', ()=> {
  it('SelectForm Snapshot', () => {
    const { asFragment } = render(<SelectForm />);
    expect(asFragment(<SelectForm />)).toMatchSnapshot()
  });
})