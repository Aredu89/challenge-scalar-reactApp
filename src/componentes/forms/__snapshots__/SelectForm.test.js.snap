// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectForm tests SelectForm Snapshot 1`] = `
<DocumentFragment>
  <div
    class="select-form w-100"
  >
    <div
      class="MuiFormControl-root makeStyles-formControl-1"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink="false"
        id="select-undefined"
      />
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="select-undefined select-undefined"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
          id="select-undefined"
          role="button"
          tabindex="0"
        >
          <span>
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-5"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
